src/main/java/codeu/controller/ActivityServlet.java

```diff
@@ -61,14 +68,21 @@ void setUserStore(UserStore userStore) {
   void setConversationStore(ConversationStore conversationStore) {
     this.conversationStore = conversationStore;
   }
+  
+  void setMentionStore(MentionStore mentionStore) {
+	this.mentionStore = mentionStore;
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
 
     List<Conversation> conversations =
         new ArrayList<Conversation>(conversationStore.getAllConversations());
+    List<Mention> mentions = 
+    	new ArrayList<Mention>(mentionStore.getAllMentions());
```

Comment:
What about moving the "filter" logic out of the JSP and into MentionStore into methods called getMentionsByUserId() and getMentionsForUserId(), and then calling those in here and setting attributes accordingly?